spec/models/candidate_interface/other_qualification_form_spec.rb

```diff
@@ -107,23 +106,20 @@
     end
   end
 
-  describe '.build_from_application' do
-    it 'returns a new OtherQualificationForm object using the id' do
-      application_form = create(:application_form) do |form|
-        form.application_qualifications.create(
-          id: 1,
-          level: 'other',
-          qualification_type: 'BTEC',
-          subject: 'Being a Sidekick',
-          institution_name: 'School of Sidekicks',
-          grade: 'Merit',
-          predicted_grade: false,
-          award_year: '2010',
-        )
-        form.application_qualifications.create(id: 2, level: 'other')
-      end
+  describe '.build_from_qualification' do
```

Comment:
:+1: 

There's another `id` left in this file which I think we could squash in another commit. On line 159 we test the update action by creating some post data including an ID, then creating an application qualification with the same id. This is probably liable to the same problems with IDs that we saw breaking in `master`. A fix might be to create the existing qualification first, then use the id in the post data.

Before

```ruby
form_data = {
  id: 1,
  qualification_type: 'BTEC',
  subject: 'Being a Everyday Hero',
  institution_name: 'School of Humans',
  grade: 'Distinction',
  award_year: '2011',
}
qualification = CandidateInterface::OtherQualificationForm.new(form_data)
application_form = create(:application_form) do |form|
  form.application_qualifications.create(
    id: 1,
    level: 'other',
    qualification_type: 'BTEC',
    subject: 'Being a Everyday Hero',
    institution_name: 'School of Hoomans',
    grade: 'Pass',
    predicted_grade: false,
    award_year: '2011',
  )
end

expect(qualification.update(application_form)).to eq(true)
expect(application_form.application_qualifications.other.first)
  .to have_attributes(form_data)
```

After 
```ruby
application_form = create(:application_form)

existing_qualification = application_form.application_qualifications.create(
  level: 'other',
  qualification_type: 'BTEC',
  subject: 'Being a Everyday Hero',
  institution_name: 'School of Hoomans',
  grade: 'Pass',
  predicted_grade: false,
  award_year: '2011',
)

form_data = {
  id: existing_qualification.id,
  qualification_type: 'BTEC',
  subject: 'Being a Everyday Hero',
  institution_name: 'School of Humans',
  grade: 'Distinction',
  award_year: '2011',
}
qualification_form = CandidateInterface::OtherQualificationForm.new(form_data)

expect(qualification_form.update(application_form)).to eq(true)
expect(application_form.application_qualifications.other.first)
  .to have_attributes(form_data)
```

Reply:
This fix makes sense to me. Thank you!

Reply:
Done in the commit about refactoring the `OtherQualificationForm` and spec.